src/ngx_http_lua_ringbuff_log.c

```diff
@@ -0,0 +1,182 @@
+
+/*
+ * Copyright (C) OpenResty, Inc.
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include <nginx.h>
+#include "ngx_http_lua_ringbuff_log.h"
+
+
+typedef struct {
+    int    is_data;
+    int    log_level;             /* attatch info */
+    size_t len;                   /* data length */
+}ringbuff_head;
```

Comment:
Also, I think `ngx_http_lua_log_ringbuf_header_t` is a better name. Please note that `head` and `header` are quite different things.